### ISSUE TITLE
Benchmark causes unknown error

### ISSUE THREAD

**Issue by mocenas** [Hyperfoil/Hyperfoil#104]
Hyperfoil v 0.7. Benchmark  [run_scenario.yaml.txt](https://github.com/Hyperfoil/Hyperfoil/files/4244448/run_scenario.yaml.txt)

Uploading it via CLI displays error:
```
Loaded benchmark write-article, uploading...
java.lang.RuntimeException: java.io.NotSerializableException: io.netty.buffer.UnpooledHeapByteBuf
```
No idea what that means.

Uploading it via HTTP API, seems that server accepted it. After starting it, agents fails after a moment. Log from one agent:
[agent.log](https://github.com/Hyperfoil/Hyperfoil/files/4244466/agent.log)

Run result is here:
[run_output.json.txt](https://github.com/Hyperfoil/Hyperfoil/files/4244447/run_output.json.txt)
see error lines:
`... agent-one: index: 182, length: 4 (expected: range(0, 182)) ...`

From these error messages, I have no idea what is wrong with this benchmark.

**Comment by rvansa** [comment]
Thanks for the report. The first issue is a bug; I found the cause and I'll prepare a fix.

The second one looks a bit as a mismatch between controller and agent version; could you add controller log to confirm it's not? I can't reproduce that myself.

I'll be doing a bugfix release this afternoon.

**Comment by rvansa** [comment]
Yep, and the work-around is using
```
- httpRequest:
     ...
     body:
       text: some text
```
I'll also make sure these both use the same (and correct) codepath.

**Comment by mocenas** [comment]
Here is a controller log:
[controller.log](https://github.com/Hyperfoil/Hyperfoil/files/4244705/controller.log)

But other scenarios works flawlesly with same controller.

**Comment by rvansa** [comment]
Okay; it looks like the benchmark was somehow partially sent before the serialization error happened (you couldn't upload due to the same problem, and when you've uploaded that using YAML upload the problem was encountered when controller uploaded it to agents).
I've locally build & pushed version `0.8-SNAPSHOT` so you can use the tag before I release `0.7.1`.

**Comment by rvansa** [comment]
I guess I've solved this in b54874fcfd4e392bf90d456496fc4b9e6f70890a
